src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.cs

```diff
@@ -102,62 +80,23 @@ public bool CreateNoWindow
 
                     // Manual use of IDictionaryEnumerator instead of foreach to avoid DictionaryEntry box allocations.
                     IDictionaryEnumerator e = envVars.GetEnumerator();
-                    try
+                    while (e.MoveNext())
                     {
-                        while (e.MoveNext())
-                        {
-                            DictionaryEntry entry = e.Entry;
-                            _environmentVariables.Add((string)entry.Key, (string)entry.Value);
-                        }
-                    }
-                    finally
-                    {
-                        (e as IDisposable)?.Dispose();
```

Comment:
@stephentoub can you explain this:
> // Manual use of IDictionaryEnumerator instead of foreach to avoid DictionaryEntry box allocations.
where would the boxing occur?